pkg/virt-handler/vm.go

```diff
@@ -207,9 +206,8 @@ func NewController(
 	})
 
 	c.launcherClients = virtcache.LauncherClientInfoByVMI{}
-	c.podInterfaceCache = virtcache.PodInterfaceByVMIAndName{}
 
-	c.networkController = netsetup.NewController(c.networkCacheStoreFactory)
+	c.networkController = netsetup.NewController(netcache.NewInterfaceCacheFactory())
```

Comment:
Why do you pass the cache factory as a parameter to the `NewController` and not initialize it inside the controller?

Reply:
Same reason it was done until now... it needs to be stubbed/mocked in the tests with one that does not use the filesystem. See [here](https://github.com/kubevirt/kubevirt/pull/6781/commits/bbbef6b221d67db58f3756a3ef7b1d411e112842#diff-d4234662480c1bd782e2ff45556a6c0d582d89cd1f012bcff7feecc14b4acfc9R213).